### PR TITLE
Flesh out metadata doc schema example

### PR DESCRIPTION
**Pull request by hyanwong**
As discussed with @benjeffery

### CODE REVIEW

File: docs/metadata.md [hunk 1]
```diff
@@ -392,22 +400,40 @@ schema = metadata.MetadataSchema(
                 "binaryFormat": "10p",
                 "pattern": "^([1-9][0-9]{3})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])?$",
             },
+            "location": {
+                "description": "latitude and longitude of the collected sample",
```
Comment (jeromekelleher) [inline docs/metadata.md:404]: Not sure this is a great example - we want people to use Individual.location for things like this, right?

**Review by jeromekelleher — COMMENTED** [review]
LGTM, but I'd choose a different example in case people think this is the recommended route for location info.

File: docs/metadata.md (hunk 1 above)
Comment (hyanwong) [reply docs/metadata.md:404]: That's a good point. Have you got a suggestion for a parameter that uses one (or a set of) floating point values that we could use?

Comment (jeromekelleher) [reply docs/metadata.md:404]: Some kind of phenotype maybe. Height and bmi or something?

Comment (hyanwong) [reply docs/metadata.md:404]: Perhaps height and sex (so that we can use an enum)?

Comment (jeromekelleher) [reply docs/metadata.md:404]: Whatever - but lets not choose something divisive as an example.

Comment (hyanwong) [reply docs/metadata.md:404]: Good point. I forget this is an arguable point.

**Comment by benjeffery** [comment]
Thanks @hyanwong, looks good modulo JK's comments.
One more detail i should comment on - `additionalProperties` is optional here as it will be added to the schema for you if you use struct. The same thing is happening with `required` which is being set on the `location` object in this case.

File: docs/metadata.md [hunk 2]
```diff
@@ -101,6 +101,15 @@ using the Python API.
 .. todo:: Add examples of adding table-level metadata schemas.
 ```
 
+### Defaults
+
+```{eval-rst}
+.. todo:: Add examples of setting default values.
+```
+
+Default values for properties can be specified for only
```
Comment (benjeffery) [inline docs/metadata.md:110]: This is only the case for JSON codec.

File: docs/metadata.md [hunk 3]
```diff
@@ -392,22 +400,41 @@ schema = metadata.MetadataSchema(
...
+                        "type": "number",
+                        "binaryFormat": "f",
+                    },
+                    "age": {
+                        "description": "Age in years at time of sampling, or -1 if unknown",
+                        "type": "number",
+                        "binaryFormat": "h",
+                    },
+                },
+                "default": {"height": float("NaN"), "age": -1},
```
Comment (benjeffery) [inline docs/metadata.md:418]: I think these should be done as separate defaults for age and height, then with an empty object default for phenotype. This allows either both, or one to be missing.

File: docs/metadata.md (hunk 2 above)
Comment (hyanwong) [reply docs/metadata.md:110]: Oh, I thought you said it was also required by the struct codec. I misunderstood. But I bunged that in a separate commit, so I can remove that. Sorry for the misunderstanding.

File: docs/metadata.md (hunk 3 above)
Comment (hyanwong) [reply docs/metadata.md:418]: Yes, I can see that, but you said previously that it wasn't obvious that you could set a top-level dict of defaults like this, so I thought it was worth showing an example.

**Comment by hyanwong** [comment]
This is now ready for merging, IMO, @benjeffery

**Comment by benjeffery** [comment]
@mergifyio rebase

**Comment by benjeffery** [comment]
Yeesh 32bit tests are failing hard. Will investigate.

**Comment by benjeffery** [comment]
@mergifyio rebase

**Review by benjeffery — APPROVED** [review]

**Pull request merged**